### PR TITLE
Persist App state + Cell state in md conversions

### PR DESCRIPTION
**Pull request by dmadisetti**
I got `marimo edit document.md` working locally and realized a few things were missing:

- [x] "Unparseable Cells" were dropped in conversion, now the "unparsable" flag is set on the markdown block. Empty cells were also dropped.
- [x] There was additional information like layout_file and width that conversion was dropping on _AppConfig
- [x] Cell Config was also dropped (disabled, show_code persists now)
- [x] Markdown cells next to each other were merged. Added a break between cells with an empty HTML comment `<!---->`

The actual code to get `marimo edit document.md` working is a bit of a hack. Once this is in, I'll open the PR for reference, but I think that it will require a deeper rework from someone more versed in the internals

### CODE REVIEW

**Comment by dmadisetti** [comment]
Oh woops. I didn't realize `|=` was newer than python 3.8. Will revise once I get home

File: marimo/_cli/convert/markdown.py [hunk 1]
```diff
@@ -2,12 +2,11 @@
...
 
 import re
-from typing import Any, Callable, Literal
+from typing import Any, Callable, Literal, Optional
 
 # Native to python
 from xml.etree.ElementTree import Element, SubElement
 
-# Note: yaml is also a python builtin
 import yaml
```
Comment (mscolnick) [inline marimo/_cli/convert/markdown.py:10]: i might have lazy imported this, just fyi, if you are not up-to-date with `main`

File: tests/_cli/snapshots/unsafe-doc.md.txt [hunk 2]
```diff
@@ -18,14 +18,48 @@ print("Hello, World!")
 
 ```marimo run convert document.md```
 
+```{.python.marimo unparsable="true"}
+it's an unparsable cell
+```
+
 <!-- Actually markdown -->
 ```{python} `
```
Comment (mscolnick) [inline tests/_cli/snapshots/unsafe-doc.md.txt:26]: is this also syntax i wasn't aware of? like an one-line code-fence?

File: marimo/_cli/convert/markdown.py [hunk 3]
```diff
@@ -42,35 +43,60 @@ def _is_code_tag(text: str) -> bool:
     return bool(re.search(r"\{.*python.*\}", head))
 
 
-def formatted_code_block(code: str) -> str:
+def formatted_code_block(
```
Comment (mscolnick) [inline marimo/_cli/convert/markdown.py:46]: sometime this week - i may write some small unit tests for these files (as my way to learn this part of the code more)

**Review by mscolnick — DISMISSED** [review]
🚢

**Comment by mscolnick** [comment]
i can force merge past playwright - its been very flakey lately and i need to investigate

File: tests/_cli/snapshots/unsafe-doc.md.txt (hunk 2 above)
Comment (dmadisetti) [reply tests/_cli/snapshots/unsafe-doc.md.txt:26]: No this is intentionally broken to make sure match works. The newest unit tests turned into more of smoke tests, so yeah- I think more granular tests would be awesome

File: marimo/_cli/convert/markdown.py (hunk 3 above)
Comment (dmadisetti) [reply marimo/_cli/convert/markdown.py:46]: 🛳️

File: marimo/_cli/convert/markdown.py (hunk 1 above)
Comment (dmadisetti) [reply marimo/_cli/convert/markdown.py:10]: I think in exporter, not here

**Comment by dmadisetti** [comment]
> i can force merge past playwright - its been very flakey lately and i need to investigate

The python 3.8 error is valid, but I won't be able to get to it for another hour or so

File: marimo/_cli/convert/markdown.py (hunk 1 above)
Comment (mscolnick) [reply marimo/_cli/convert/markdown.py:10]: oh thats right

File: tests/_cli/snapshots/unsafe-doc.md.txt (hunk 2 above)
Comment (mscolnick) [reply tests/_cli/snapshots/unsafe-doc.md.txt:26]: oh gotcha, ha makes more sense

**Comment by mscolnick** [comment]
@dmadisetti there is a snapshot fail on `Python 3.11` for some reason

**Comment by dmadisetti** [comment]
Oh woops. Weird, that's what I run

**Comment by mscolnick** [comment]
thanks @dmadisetti for more great work!

**Pull request merged**